CMakeLists.txt

```diff
@@ -130,9 +130,28 @@ endif()
 
 set(DEFAULT_GPU_ARCH "" CACHE STRING "Optional: Default GPU architecture to compile for when targeting GPUs (e.g.: sm_60 or gfx900)")
 set(ROCM_LIBS "-L${ROCM_PATH}/lib -L${ROCM_PATH}/hip/lib -lamdhip64" CACHE STRING "Necessary libraries for ROCm")
-set(ROCM_LINK_LINE "-rpath ${ROCM_PATH}/lib -rpath ${ROCM_PATH}/hip/lib ${ROCM_LIBS}")
+if(NOT ROCM_LINK_LINE)
+  set(ROCM_LINK_LINE "-rpath ${ROCM_PATH}/lib -rpath ${ROCM_PATH}/hip/lib ${ROCM_LIBS}")
+endif()
+if(NOT ROCM_CXX_FLAGS)
+  set(ROCM_CXX_FLAGS "-I$HIPSYCL_ROCM_PATH/include -I$HIPSYCL_ROCM_PATH/include --hip-device-lib-path=$HIPSYCL_ROCM_PATH/lib")
+endif()
 set(CUDA_LINK_LINE "-rpath $HIPSYCL_CUDA_LIB_PATH -L$HIPSYCL_CUDA_LIB_PATH -lcudart" CACHE STRING "Arguments passed to compiler to link CUDA libraries to SYCL applications")
-
+if(NOT CUDA_CXX_FLAGS)	
+  set(CUDA_CXX_FLAGS "")
+endif()
+if(NOT OMP_LINK_LINE)
+  set(OMP_LINK_LINE "-L${BOOST_PATH}/lib -rpath ${BOOST_PATH}/lib")
```

Comment:
I think we should also add `-fopenmp` here (users may want to cutomize this e.g. to `-fopenmp=gomp` if they want to use GNU OpenMP with clang).

I'm not sure if `boost_context`  and `boost_fiber` should be added as well?

We are now using `-Wl,-rpath=` inside `syclcc`. Also use here?

Reply:
I think adding more should not be a problem, I will also add the `-Wl,-rpath=` I am not sure why this works :D